src/System.Text.Json/src/System/Text/Json/JsonDocument.cs

```diff
@@ -326,7 +330,8 @@ internal bool TryGetValue(int index, out double value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (Utf8JsonReader.TryGetDoubleValue(segment, out double tmp))
+            char standardFormat = segment.IndexOfAny((byte)'e', (byte)'E') >= 0 ? 'e' : default;
+            if (Utf8Parser.TryParse(segment, out double tmp, out int bytesConsumed, standardFormat) && segment.Length == bytesConsumed)
```

Comment:
Since the value is `out` instead of `ref`, and it's not an exception, it doesn't really matter. (Technically it's different if you out to a field and more than one thread reads the field, since the `out` is set directly, instead of when the method returns... but that just means the caller needs to improve their locking)